test/metabase/util_test.cljc

```diff
@@ -508,3 +509,21 @@
        (let [expected (reduce + (u/map-all (fnil + 0 0 0) xs ys zs))
              sum (+ (reduce + 0 xs) (reduce + 0 ys) (reduce + 0 zs))]
          (= expected sum)))))
+
+#?(:clj
+   (deftest ^:parallel case-enum-test
+     (testing "case does not work"
+       (is (not (case Month/MAY
+                  Month/APRIL false
+                  Month/MAY   true
+                  false))))
+     (testing "case-enum works"
+       (is (u/case-enum Month/MAY
+             Month/APRIL false
+             Month/MAY   true
+             false)))
+     (testing "checks for type"
+       (is (thrown? Exception #"`case-enum` only works.*"
```

Comment:
It's great that we check this, but it still leaves the slightly more likely bug which is having a mismatch just in runtime value we take, e.g.

```clojure
(u/case-enum DayOfWeek/SUNDAY
  Month/JANUARY 1)
```

We can't protect against this without reflection or a more advanced technique, so I think this is fine, but perhaps worth a test documenting the sad as well.